thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/rootcause/impl/HolidayEventsLoader.java

```diff
@@ -0,0 +1,142 @@
+   *
+   * @param configuration the configuration
+   */
+  public HolidayEventsLoader(ThirdEyeAnomalyConfiguration configuration) {
+    calendarList = configuration.getCalendars();
+    keyPath = configuration.getRootDir() + configuration.getKeyPath();
+    scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+  }
+
+  /** List of google holiday calendar ids */
+  private static List<String> calendarList;
+  private static String keyPath;
+  private ScheduledExecutorService scheduledExecutorService;
+  private TimeGranularity runFrequency = new TimeGranularity(7, TimeUnit.DAYS);
+
+  private static final Logger LOG = LoggerFactory.getLogger(HolidayEventsLoader.class);
+
+  /** Global instance of the HTTP transport. */
+  private static HttpTransport HTTP_TRANSPORT;
+
+  /** Global instance of the JSON factory. */
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+  /** Global instance of the scopes. */
+  private static final Set<String> SCOPES = Collections.singleton(CalendarScopes.CALENDAR_READONLY);
+
+  static {
+    try {
+      HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+    } catch (Exception e) {
+      LOG.error("Can't create http transport with google api.");
+    }
+  }
+
+  private static final DAORegistry DAO_REGISTRY = DAORegistry.getInstance();
+
+  /**
+   * Start.
+   */
+  public void start() {
+    scheduledExecutorService.scheduleAtFixedRate(this, 0, runFrequency.getSize(), runFrequency.getUnit());
+  }
+
+  /**
+   * Shutdown.
+   */
+  public void shutdown() {
+    scheduledExecutorService.shutdown();
+  }
+
+  /**
+   * Fetch holidays and save to ThirdEye database.
+   */
+  public void run() {
+    EventManager eventDAO = DAO_REGISTRY.getEventDAO();
+
+    List<Event> holidays = getAllHolidays();
+    Set<EventDTO> newEvents = new HashSet<>();
+
+    for (Event holiday : holidays) {
+      EventDTO eventDTO = new EventDTO();
+
+      eventDTO.setStartTime(holiday.getStart().getDate().getValue());
+      eventDTO.setEndTime(holiday.getEnd().getDate().getValue());
+      eventDTO.setName(holiday.getSummary());
+      eventDTO.setEventType(EventType.HOLIDAY.toString());
+
+      newEvents.add(eventDTO);
+    }
+    Set<EventDTO> existingEvents = new HashSet<>(eventDAO.findAll());
+
+    // remove existing holidays
+    newEvents.removeAll(existingEvents);
+
+    // add new holidays
+    for (EventDTO newEvent : newEvents) {
+      eventDAO.save(newEvent);
+    }
+  }
+
+  /**
+   * Fetch all holidays from Google Calendar API.
+   */
+  private List<Event> getAllHolidays() {
+    List<Event> events = new ArrayList<>();
+    for (String calendar : calendarList) {
+      try {
+        events.addAll(this.getCalendarEvents(calendar));
+      } catch (Exception e) {
+        // left blank
+      }
+    }
+    return events;
+  }
+
+  private List<Event> getCalendarEvents(String Calendar_id) throws Exception {
+
```

Comment:
Throw an exception if HTTP_TRANSPORT is not initialized successfully at line 67.